### ISSUE TITLE
Tomcat monitoring issues with Casper MDM

### ISSUE THREAD

**Issue by GitStoph** [firehol/netdata#1798]
Firstly, love this project. Hope you guys can help me!

`/usr/libexec/netdata/plugins.d/python.d.plugin debug 1 tomcat`
yields:
2017-02-15 16:53:07:  INFO: Using python v2
2017-02-15 16:53:07:  INFO: YAML output is ordered
2017-02-15 16:53:07: python.d INFO: reading configuration file: /etc/netdata/python.d.conf
2017-02-15 16:53:07: python.d ERROR: /etc/netdata/python.d.conf is not a valid configuration file
2017-02-15 16:53:07: python.d INFO: MODULES_DIR='/usr/libexec/netdata/python.d/', CONFIG_DIR='/etc/netdata/', UPDATE_EVERY=1, ONLY_MODULES=['tomcat']
2017-02-15 16:53:07: python.d DEBUG: tomcat: loading module configuration: '/etc/netdata/python.d/tomcat.conf'
2017-02-15 16:53:07: python.d ERROR: /etc/netdata/python.d/tomcat.conf is not a valid configuration file
2017-02-15 16:53:07: python.d DEBUG: tomcat: reading configuration
2017-02-15 16:53:07: python.d DEBUG: tomcat: job added
2017-02-15 16:53:07: python.d DEBUG: all job objects [<Service(None, initial daemon)>]
2017-02-15 16:53:07: python.d ERROR: tomcat_local <urlopen error [Errno 111] Connection refused>
2017-02-15 16:53:07: python.d ERROR: tomcat_local check() failed - disabling job
2017-02-15 16:53:07: python.d ERROR: DISABLED: tomcat/local
2017-02-15 16:53:07: python.d DEBUG: overridden job names: []
2017-02-15 16:53:07: python.d DEBUG: all remaining job objects: []
2017-02-15 16:53:07: python.d FATAL: no more jobs
DISABLE

`python --version`
Python 2.7.12

`/etc/netdata/python.d/tomcat.conf`
tomcat1:
url : 'https://my.cool.domain:8443'
user : 'tomcat'
pass : '12345'

` curl -u tomcat:12345 https://my.cool.domain:8443`

<!DOCTYPE html>

<html lang="en" class="no-js" ng-app="jamf.login"> 

<head>
  <meta http-equiv="Pragma" content="no-cache">
  <meta http-equiv="expires" content="0">
  <meta HTTP-EQUIV="content-type" CONTENT="text/html;charset=utf-8">
  <meta charset="utf-8" />
  <meta name="description" content="" />
  <meta name="author" content="JAMF Software" />
  <meta name="apple-mobile-web-app-title" content="JSS">
  <meta name="version" content="9.97.1482356336">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
  <title>JSS Login - JSS v9.97.1482356336</title>

  <script type="text/javascript" id="setBaseHrefScript">
    var a = document.createElement('a');
    a.href = window.location + '';

    var path = a.pathname; // defaults to '/' if not set in window.location

    // Except in the case of IE, it may need '/' to be prepended.
    if (path.slice(0, 1) !== '/') {
        path = '/' + path;
    }
    var components = path.split('/');
    var context = '/';

    if(components.length > 1 && components[1] && components[1] !== 'view' && ! components[1].match(/^(.*\.html)/)) {
        context += components[1] + '/';
        console.log('using application context ' + context);
    }

    var base = document.createElement("base");
    base.setAttribute("href", context);
    var script = document.getElementById("setBaseHrefScript");
    script.parentNode.insertBefore(base, script);
</script>

  <script type="text/javascript" src="javascripts/jquery/jquery.js"></script>
  <script type="text/javascript" src="javascripts/scripts.js"></script>
  <script type="text/javascript" src="javascripts/jquery.tablesorter.min.js"></script>

  
  <link rel="apple-touch-icon-precomposed" href="ui/images/touchicons/apple-touch-icon-precomposed.png">
  
  <link rel="apple-touch-icon-precomposed" sizes="72x72" href="ui/images/touchicons/apple-touch-icon-72x72-precomposed.png">
  
  <link rel="apple-touch-icon-precomposed" sizes="114x114" href="ui/images/touchicons/apple-touch-icon-114x114-precomposed.png">
  
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="ui/images/touchicons/apple-touch-icon-144x144-precomposed.png">
  <link rel='mask-icon' color="#8DC63F" href="ui/images/touchicons/JSS_pinned.svg" >

  
<link rel="stylesheet" href="ui/styles/main.css" />
<link rel="stylesheet" href="ui/styles/legacy.css" />
<link rel="stylesheet" href="ui/libs/ui-grid.css" />

<script src="ui/jamf.js?v=1482368957778"></script>

</head>

<body id="wrapper">
  <div id="emptyVerticalCenteringFloat"></div>
  <div id="login-wrapper" ng-controller="loginController as lc">
    <form class="loginForm" method="POST" name="f">
      <div id="login-panel">
        <div id="login-icon"></div>
        <div id="login-fields">
          <span class="label">Username</span>
          <input name="username" id="username" class="input" value="" autocapitalize="off" autocorrect="off"
               ng-bind="lc.username" />

          <span class="label">Password</span>
          <input name="password" id="password" class="input" type="password" autocomplete="off"
               ng-bind="lc.password" />

          <input type="submit" class="button" value="Log In" id="jssLoginButton" />

        </div>

        <div id="reset-fields" style="display:none">
          <span class="label">Username</span>
          <input name="resetUsername" id="resetUsername" class="input" value="" autocapitalize="off" autocorrect="off"/>

          <span class="label">&nbsp;</span>
          <input type="hidden" disabled="disabled" name="reset" value="reset" />
          <input type="submit" class="button" value="Reset Password" id="reset" />
        </div>
      </div>
    </form>
    <copyright></copyright>
  </div>
</body>

<script>
function setDefaultField(){

  document.f.username.focus();

}

function resetPassword(){
  $("#login-fields").hide();
  $("#reset-fields").show();
  $("#reset-fields input[name='reset']").removeAttr("disabled");
}
$("#reset").click(function() {
  //resend the reset field value
  $(".loginForm").submit();
});
</script>

Basically the curl results the original login page.. However the username and password will log you into the application correctly. Is the problem going to be the way casper has the page coded? Or any ideas?

**Comment by ktsaou** [comment]
I see the plugin expects the default port to be 8080: https://github.com/firehol/netdata/blob/b081a001f8181ce34fbd579a9b7b278cdcd4c786/python.d/tomcat.chart.py#L49

You can configure it: https://github.com/firehol/netdata/blob/b081a001f8181ce34fbd579a9b7b278cdcd4c786/conf.d/python.d/tomcat.conf#L73

But you need to give the full URL: `http://tomcat:8443/manager/status?XML=true`

To check it, check exactly the same URL with `curl`.

**Comment by ktsaou** [comment]
Keep in mind the log says:

```
2017-02-15 16:53:07: python.d ERROR: tomcat_local <urlopen error [Errno 111] Connection refused>
```

So, it cannot connect at all to the URL you have configured.

@l2isbad it would be nice to log the URL too in the log, just to make sure we are checking the right thing.

**Comment by l2isbad** [comment]
@GitStoph

And

2017-02-15 16:53:07: python.d ERROR: /etc/netdata/python.d/**tomcat.conf is not a valid configuration file**

**Comment by GitStoph** [comment]
Good morning!
@l2isbad 
I took a sanitized tomcat.conf file and added the lines at the very end without compromising anything else. Is that incorrect? I was attempting to follow a previous issue 17something (sorry, not at work today, will definitely get that reference correct later!!)

So JAMF's Casper MDM takes their custom ROOT.war to be dropped into webapps. From all the digging I've done, the status?XML=true page does not exist. Digging their their forums lead me to this non-monitoring-helpful item: http://docs.jamf.com/9.9/casper-suite/administrator-guide/JSS_Health_Check_Page.html

I was also having issues using the web_log plugin with casper's stuff as they do not logrotate nicely. every DAY has a different access log in localhost_year_month_day.log format. I can create a separate issue for that when I'm back to work and have done more documented troubleshooting though. 

@ktsaou 
Thanks for pointing that line out, I'll attempt to change that to see if it helps. I saw that it was refusing the connection, but I thought it was reading my URL correctly. I guess I need to better understand what it's not liking about that conf file when I followed a format that others had used from other issues? 

MDMs tend to be suuuper chatty and fickle little things. It'd be amazing to get those netdata plugins working on it! Thanks!

**Comment by l2isbad** [comment]
@GitStoph you need not change anything. The only thing tomcat plugin expect is `url` and `url` must endswith `manager/status?XML=true` `otherwise tomcat_local check() failed - disabling job`.

Current `web_log` plugin version works only with apache/nginx/gunicorn/lighttpd like logs.

Can you post here some `JAMF's Casper MDM` (btw name looks cool) logs?

**Comment by GitStoph** [comment]
Good morning,
So each access log uses the format localhost_access_log.YEAR-MO-DA.txt
Here is an example snippit:
10.147.16.215 - - [17/Feb/2017:10:21:01 -0600] "POST /client HTTP/1.1" 200 2656
10.147.16.215 - - [17/Feb/2017:10:21:03 -0600] "POST /client HTTP/1.1" 200 3086
10.147.16.215 - - [17/Feb/2017:10:21:07 -0600] "POST /client HTTP/1.1" 200 1050
206.131.134.51 - - [17/Feb/2017:10:22:06 -0600] "POST /client HTTP/1.1" 200 2712
206.131.134.51 - - [17/Feb/2017:10:22:11 -0600] "POST /client HTTP/1.1" 200 3141
10.8.14.46 - - [17/Feb/2017:10:24:15 -0600] "POST /client HTTP/1.1" 200 2656
10.8.14.46 - - [17/Feb/2017:10:24:17 -0600] "POST /client HTTP/1.1" 200 3085
10.8.14.46 - - [17/Feb/2017:10:24:20 -0600] "POST /client HTTP/1.1" 200 1050
10.1.4.40 - - [17/Feb/2017:10:24:38 -0600] "POST /client HTTP/1.1" 200 2656
10.1.4.40 - - [17/Feb/2017:10:24:40 -0600] "POST /client HTTP/1.1" 200 3084
10.2.1.10 - - [17/Feb/2017:10:26:01 -0600] "POST /client HTTP/1.1" 200 2656
10.2.1.10 - - [17/Feb/2017:10:26:03 -0600] "POST /client HTTP/1.1" 200 3084
10.2.1.10 - - [17/Feb/2017:10:26:07 -0600] "POST /client HTTP/1.1" 200 1050
10.2.1.12 - - [17/Feb/2017:10:28:44 -0600] "POST /client HTTP/1.1" 200 2712
10.2.1.12 - - [17/Feb/2017:10:28:46 -0600] "POST /client HTTP/1.1" 200 3144
10.2.1.12 - - [17/Feb/2017:10:28:49 -0600] "POST /client HTTP/1.1" 200 1106
10.138.17.76 - - [17/Feb/2017:10:30:32 -0600] "POST /client HTTP/1.1" 200 2712
10.138.17.76 - - [17/Feb/2017:10:30:33 -0600] "POST /client HTTP/1.1" 200 3138
10.138.17.76 - - [17/Feb/2017:10:30:36 -0600] "POST /client HTTP/1.1" 200 2712
10.138.17.76 - - [17/Feb/2017:10:30:38 -0600] "POST /client HTTP/1.1" 200 3138
10.2.1.11 - - [17/Feb/2017:10:31:13 -0600] "POST /client HTTP/1.1" 200 2712
10.2.1.11 - - [17/Feb/2017:10:31:15 -0600] "POST /client HTTP/1.1" 200 3140
10.2.1.11 - - [17/Feb/2017:10:31:18 -0600] "POST /client HTTP/1.1" 200 1106
206.131.134.51 - - [17/Feb/2017:10:32:13 -0600] "POST /client HTTP/1.1" 200 2712
206.131.134.51 - - [17/Feb/2017:10:32:17 -0600] "POST /client HTTP/1.1" 200 3141
206.131.134.51 - - [17/Feb/2017:10:32:22 -0600] "POST /client HTTP/1.1" 200 1106
10.2.1.13 - - [17/Feb/2017:10:32:38 -0600] "POST /client HTTP/1.1" 200 2712
10.2.1.13 - - [17/Feb/2017:10:32:40 -0600] "POST /client HTTP/1.1" 200 3140
10.2.1.13 - - [17/Feb/2017:10:32:43 -0600] "POST /client HTTP/1.1" 200 1106
10.10.20.215 - - [17/Feb/2017:10:34:18 -0600] "POST /client HTTP/1.1" 200 2712
10.10.20.215 - - [17/Feb/2017:10:34:21 -0600] "POST /client HTTP/1.1" 200 3136
10.10.20.215 - - [17/Feb/2017:10:34:24 -0600] "POST /client HTTP/1.1" 200 1106
10.126.14.178 - - [17/Feb/2017:10:35:27 -0600] "POST /client HTTP/1.1" 200 2656
10.126.14.178 - - [17/Feb/2017:10:35:30 -0600] "POST /client HTTP/1.1" 200 3084
10.126.14.178 - - [17/Feb/2017:10:35:34 -0600] "POST /client HTTP/1.1" 200 1050
206.131.134.51 - - [17/Feb/2017:10:36:08 -0600] "POST /client HTTP/1.1" 200 2712
206.131.134.51 - - [17/Feb/2017:10:36:13 -0600] "POST /client HTTP/1.1" 200 3141
10.10.20.215 - - [17/Feb/2017:10:36:16 -0600] "POST /client HTTP/1.1" 200 2712
10.10.20.215 - - [17/Feb/2017:10:36:16 -0600] "PUT //computer/mdm HTTP/1.1" 200 -
10.10.20.215 - - [17/Feb/2017:10:36:16 -0600] "POST /client HTTP/1.1" 200 2712
10.10.20.215 - - [17/Feb/2017:10:36:18 -0600] "POST /client HTTP/1.1" 200 3136
10.10.20.215 - - [17/Feb/2017:10:36:21 -0600] "POST /client HTTP/1.1" 200 1106

**Comment by GitStoph** [comment]
http://resources.jamfsoftware.com/documents/white-papers/Securing-Your-JSS.pdf
Also located this. Looks like the log format could be formatted. 
`<Valve className=”org.apache.catalina.valves.AccessLogValve” directory=”logs”
prefix=”localhost_access_log.” suffix=”.txt”
pattern=”%h %l %u %t &quot;%r&quot; %s %b” resolveHosts=”false”/>`

**Comment by l2isbad** [comment]
> Looks like the log format could be formatted

@GitStoph the log format is ok (same as apache) for web_log plugin

The problem is that a new file is created each day

> 10.147.16.215 - - [17/Feb/2017:10:21:01 -0600] "POST /client HTTP/1.1" 200 2656

https://github.com/firehol/netdata/blob/master/python.d/web_log.chart.py#L276-L280

```python
re.search(r'(?P<address>[\da-f.:]+)'
                r' -.*?"(?P<method>[A-Z]+)'
                r' (?P<url>.*?)"'
                r' (?P<code>[1-9]\d{2})'
                r' (?P<bytes_sent>\d+|-)', '10.147.16.215 - - [17/Feb/2017:10:21:01 -0600] "POST /client HTTP/1.1" 200 2656').groupdict()

{'address': '10.147.16.215',
 'bytes_sent': '2656',
 'code': '200',
 'method': 'POST',
 'url': '/client HTTP/1.1'}
```

**Comment by GitStoph** [comment]
Good morning,
Would you prefer this be done as a separate issue/request? Let me know and I'll make a new one instead!

Wondering if my syntax is incorrect for the web_log plugin? 
`# ubuntu`
`apache_log4:`
  `name: 'apache'`
  `path: '/var/log/tomcat7/localhost_access_log.2017-02-21.txt'`

When I run `/usr/libexec/netdata/plugins.d/python.d.plugin debug 1 web_log` , I keep getting the following response: (I used a sanitized config and added the above values exactly)

2017-02-21 10:16:45:  INFO: Using python v2
2017-02-21 10:16:45:  INFO: YAML output is ordered
2017-02-21 10:16:45: python.d INFO: reading configuration file: /etc/netdata/python.d.conf
2017-02-21 10:16:45: python.d ERROR: /etc/netdata/python.d.conf is not a valid configuration file
2017-02-21 10:16:45: python.d INFO: MODULES_DIR='/usr/libexec/netdata/python.d/', CONFIG_DIR='/etc/netdata/', UPDATE_EVERY=1, ONLY_MODULES=['web_log']
2017-02-21 10:16:45: python.d DEBUG: web_log: loading module configuration: '/etc/netdata/python.d/web_log.conf'
2017-02-21 10:16:45: python.d ERROR: /etc/netdata/python.d/web_log.conf is not a valid configuration file
2017-02-21 10:16:45: python.d DEBUG: web_log: reading configuration
2017-02-21 10:16:45: python.d DEBUG: web_log: job added
2017-02-21 10:16:45: python.d DEBUG: all job objects [<Service(None, initial daemon)>]
2017-02-21 10:16:45: python.d ERROR: web_log log path is not specified
2017-02-21 10:16:45: python.d ERROR: web_log check() failed - disabling job
2017-02-21 10:16:45: python.d ERROR: DISABLED: web_log/None
2017-02-21 10:16:45: python.d DEBUG: overridden job names: []
2017-02-21 10:16:45: python.d DEBUG: all remaining job objects: []
2017-02-21 10:16:45: python.d FATAL: no more jobs
DISABLE

Please advise? Thanks much!

**Comment by l2isbad** [comment]
> Wondering if my syntax is incorrect for the web_log plugin? Does the log file have to end with .log?

no

I see the problem here

2017-02-21 10:16:45: python.d DEBUG: web_log: loading module configuration: '**/etc/netdata/python.d/web_log.conf**'
2017-02-21 10:16:45: python.d ERROR: /etc/netdata/python.d/web_log.conf **is not a valid** configuration file

**Comment by GitStoph** [comment]
Ok, so what correction is necessary to the file to make it valid? I used a sanitized conf file and just appended that to the end. I guess my untrained eyes aren't realizing what my syntax error is?
Sorry, thanks for the response!

**Comment by l2isbad** [comment]
@GitStoph you always can copy valid configuration from orig conf file (`module_name.conf.orig`)

```yaml
# ubuntu
apache_log4:
    name: 'apache'
    path: '/var/log/tomcat7/localhost_access_log.2017-02-21.txt'
```

**Comment by GitStoph** [comment]
@l2isbad Just tried this. The orig conf file itself is being reported as invalid. I've also tried completely removing/uninstalling netdata and reinstalling. Is there any other info I could get that'd help?

Running: Ubuntu-Server 16.04 LTS _Xenial Xerus_ - Release amd64 (20160420.3)
python: Python 2.7.12

`/etc/netdata/python.d$ /usr/libexec/netdata/plugins.d/python.d.plugin debug 1 web_log` yields:

2017-02-22 09:32:06:  INFO: Using python v2
2017-02-22 09:32:06:  INFO: YAML output is ordered
2017-02-22 09:32:06: python.d INFO: reading configuration file: /etc/netdata/python.d.conf
2017-02-22 09:32:06: python.d ERROR: /etc/netdata/python.d.conf is not a valid configuration file
2017-02-22 09:32:06: python.d INFO: MODULES_DIR='/usr/libexec/netdata/python.d/', CONFIG_DIR='/etc/netdata/', UPDATE_EVERY=1, ONLY_MODULES=['web_log']
2017-02-22 09:32:06: python.d DEBUG: web_log: loading module configuration: '/etc/netdata/python.d/web_log.conf'
2017-02-22 09:32:06: python.d ERROR: /etc/netdata/python.d/web_log.conf is not a valid configuration file
2017-02-22 09:32:06: python.d DEBUG: web_log: reading configuration
2017-02-22 09:32:06: python.d DEBUG: web_log: job added
2017-02-22 09:32:06: python.d DEBUG: all job objects [<Service(None, initial daemon)>]
2017-02-22 09:32:06: python.d ERROR: web_log log path is not specified
2017-02-22 09:32:06: python.d ERROR: web_log check() failed - disabling job
2017-02-22 09:32:06: python.d ERROR: DISABLED: web_log/None
2017-02-22 09:32:06: python.d DEBUG: overridden job names: []
2017-02-22 09:32:06: python.d DEBUG: all remaining job objects: []
2017-02-22 09:32:06: python.d FATAL: no more jobs
DISABLE

**Comment by l2isbad** [comment]
@GitStoph 
Well, do you have any `python.d.conf ` in `/etc/netdata/` or `web_log.conf` in  `/etc/netdata/python.d/` ?

**Comment by l2isbad** [comment]
[CONFIG_DIR is '/etc/netdata'](https://github.com/firehol/netdata/blob/master/plugins.d/python.d.plugin#L24) if there is no NETDATA_CONFIG_DIR env var.

Since you the command from your shell NEDATA_CONFIG_DIR is missing.

**Comment by GitStoph** [comment]
@l2isbad 
So, this is interesting. I did an update && upgrade followed by another removal/reinstall of netdata and configured the web_log.conf ( which yes, was located in `/etc/netdata/python.d/web_log.conf`). Also, the python.d.conf is in /etc/netdata . I used the following conf:
`# ubuntu`
`apache_log4:`
`  name: 'apache'`
`  path: '/var/log/tomcat7/localhost_access_log.2017-02-22.txt'`

When I run the debug I still get:
2017-02-22 09:58:58:  INFO: Using python v2
2017-02-22 09:58:58:  INFO: YAML output is ordered
2017-02-22 09:58:58: python.d INFO: reading configuration file: /etc/netdata/python.d.conf
2017-02-22 09:58:58: python.d ERROR: /etc/netdata/python.d.conf is not a valid configuration file
2017-02-22 09:58:58: python.d INFO: MODULES_DIR='/usr/libexec/netdata/python.d/', CONFIG_DIR='/etc/netdata/', UPDATE_EVERY=1, ONLY_MODULES=['web_log']
2017-02-22 09:58:58: python.d DEBUG: web_log: loading module configuration: '/etc/netdata/python.d/web_log.conf'
2017-02-22 09:58:58: python.d ERROR: /etc/netdata/python.d/web_log.conf is not a valid configuration file
2017-02-22 09:58:58: python.d DEBUG: web_log: reading configuration
2017-02-22 09:58:58: python.d DEBUG: web_log: job added
2017-02-22 09:58:58: python.d DEBUG: all job objects [<Service(None, initial daemon)>]
2017-02-22 09:58:58: python.d ERROR: web_log log path is not specified
2017-02-22 09:58:58: python.d ERROR: web_log check() failed - disabling job
2017-02-22 09:58:58: python.d ERROR: DISABLED: web_log/None
2017-02-22 09:58:58: python.d DEBUG: overridden job names: []
2017-02-22 09:58:58: python.d DEBUG: all remaining job objects: []
2017-02-22 09:58:58: python.d FATAL: no more jobs
DISABLE

HOWEVER, when I go to the netdata page for this server, web log apache IS actually showing up and displaying information now! Now I'm smacking my forehead and wondering how long it was actually showing correctly, but I wasn't reloading the actual netdata page since I was watching the debug report's errors/disabling..

**Comment by l2isbad** [comment]
@GitStoph 
It's kinda bug  - CONFIG_DIR is always '/etc/netdata/' when you run python.d.plugin debug from shell.

**Comment by GitStoph** [comment]
For anyone that may end up stumbling on this thread in the future that has issues with webservers with frequently changing log files, I created the following script that so far seems to work perfectly with updating the web_log.conf file, which I'll add to a cronjob to run at 12:05AM every day. You may of course want to remove the extra junk, I just left it in to verify my variables were registering correctly.
```
#!/bin/sh
#
#2.22.17 CJM
#web_log plugin editing for Netdata
###########################################################
#Variables
date=`date +%Y-%m-%d`
yesterday=`date +%Y-%m-%d -d "yesterday"`
oldlog=$(echo /var/log/tomcat7/localhost_access_log.$yesterday.txt)
newlog=$(echo /var/log/tomcat7/localhost_access_log.$date.txt)
weblog=/etc/netdata/python.d/web_log.conf

# Checking if variables are working.
echo -----------------------------------------------------
echo Today $date
echo Yesterday $yesterday
echo The old log is $oldlog
echo The new log will be $newlog
echo Editing $weblog
echo -----------------------------------------------------

# Killing the netdata process
echo -----------------------------------------------------
echo Killing netdata to perform the update.
echo -----------------------------------------------------
sudo killall netdata
sleep 3

# Attempting to edit the conf.
echo -----------------------------------------------------
echo Editing $weblog
echo -----------------------------------------------------
find $weblog -type f -exec sed -i "s#$oldlog#$newlog#" {} \;
sleep 3

# Restarting the netdata process.
echo -----------------------------------------------------
echo Restarting netdata process.
echo -----------------------------------------------------
sudo /usr/sbin/netdata

echo -----------------------------------------------------
echo All done.
echo -----------------------------------------------------

exit 0
```

**Comment by ktsaou** [comment]
nice! so is there anything we can do here? Shall we close it?

**Comment by GitStoph** [comment]
@ktsaou - I leave that up to you sir. The only thing I could add to this that'd potentially assist others would be to document the setup of FireQOS on Ubuntu with a setup/config customized towards Casper's needs. I've begun to go through it and have a working config, just needs a bit more customization. If you'd like, I could post it here when I'm done for potential others to follow.

**Comment by ktsaou** [comment]
sounds good. Looking forward for it... :+1:

**Comment by GitStoph** [comment]
Good afternoon,

So after installing on ubuntu and following the instructions on https://firehol.org/tutorial/fireqos-new-user/ , I made a rookie mistake. followed the instructions verbatim and didn't realize that my conf file that it was using actually resided in `/etc/fireqos/fireqos.conf` . After that things were much more straight forward.

```
# FireQOS configuration file
#
# See fireqos.conf(5) manual page and FireQOS Manual for details.
#
# This configuration file is for illustration purpose only and
# will shape no traffic.
DEVICE=eth0
INPUT_SPEED=1Gbit
OUTPUT_SPEED=1Gbit

interface $DEVICE world-in input rate $INPUT_SPEED balanced
   class apple
      match tcp sports 5223         # <<< Apple's push notifications
   class caspertomcat
      match tcp port 8443           # <<< Most common tomcat port for JSS
   class mysql
      match port 3306               # <<< From JSS to MySQL default
   class casperdistpoint
      match port 443                # <<< push proxy patch servers etc
   class AFP
      match port 548                # <<< Used for AFP file share dist points
   class SMB
      match port 139                # <<< Used for SMB file share dist points
   class LDAP
      match port 389,636            # <<< Used for LDAP and sec LDAP
   class caspertoapns
      match port 2195               # <<< Send messages from JSS to APNS
   class casperapnsfeedback
      match port 2196               # <<< JSS to connect to APNS for feedback
   class casperqueuemessages
      match port 61617              # <<< Used to queue and dequeue messages from the message broker
   class netdata
      match port 19999              # <<< Netdata configured port
   class interactive
      match udp port 53             # <<< DNS
      match tcp port 22             # <<< SSH
      match icmp                    # <<< ping

interface $DEVICE world-out output rate $OUTPUT_SPEED balanced
   class apple
      match tcp sports 5223         # <<< Apple's push notifications
   class caspertomcat
      match tcp port 8443           # <<< Most common tomcat port for JSS
   class mysql
      match port 3306               # <<< From JSS to MySQL default
   class casperdistpoint
      match port 443                # <<< push proxy patch servers etc
   class AFP
      match port 548                # <<< Used for AFP file share dist points
   class SMB
      match port 139                # <<< Used for SMB file share dist points
   class LDAP
      match port 389,636            # <<< Used for LDAP and sec LDAP
   class caspertoapns
      match port 2195               # <<< Send messages from JSS to APNS
   class casperapnsfeedback
      match port 2196               # <<< JSS to connect to APNS for feedback
   class casperqueuemessages
      match port 61617              # <<< Used to queue and dequeue messages from the message broker
   class netdata
      match port 19999              # <<< Netdata configured port
   class interactive
      match udp port 53             # <<< DNS
      match tcp port 22             # <<< SSH
      match icmp                    # <<< ping
```

Naturally it bothered me that it said only 10000kbit would be committed per class, but then I went on to realize that the spare bandwidth would get equally applied to any class in need. Anyhoo, now between QoS, web_log, and MySQL plugins, the MDM feels quite adequately monitored! Thanks everyone for your help!
